src/SAML2/XML/saml/AttributeValue.php

```diff
@@ -31,25 +37,13 @@ class AttributeValue extends AbstractSamlElement
      *  - string
      *  - int
      *  - null
-     *  - \SimpleSAML\XML\AbstractElement[]
```

Comment:
I believe the array was a mistake. It doesn't make sense to add multiple structures in one value if you can add multiple AttributeValues. Also, the default for minOccurs/maxOccurs in XML in `1` and that should imo also count for xs:anyType

Reply:
It might be. The way I always understood the spec, you could have whatever in there, and that includes XML with multiple elements without a clear root (except for the `AttributeValue` element itself, that is). Not sure if that would make much sense since the idea of putting XML there would be that it works standalone, rather than depending on the `AttributeValue` where it is contained. And I definitely agree that it doesn't make much sense to have multiple elements inside a single `AttributeValue`, when you can just have multiple `AttributeValue` elements, one per each child.

So even if my understanding of the spec is right, I agree that it doesn't make much sense to support this scenario, specially if it makes our life harder.